src/modules/user/user.controller.ts

```diff
@@ -1,5 +1,37 @@
+import { Request } from "express";
+import { addUser, findByEmail } from "./user.dal";
+import { Login, Signup } from "./user.interfaces";
+import bcrypt from 'bcrypt'
+
 class UserController {
     constructor() { }
+    loginUser = async (req: Request) => {
+         try {
+             const data:Login={email:req.body.email,password:req.body.password}
+             const result =await findByEmail(data)
+             return result
+         } catch (error) {
+             
+         }
+    }
+    signUpUser = async (req: Request) => {
+        req.body.passwordHash = bcrypt.hashSync(req.body.password, 5)
+        console.log(req.body.passwordHash);
+        
+        try {
+            const data: Signup = {
+                name: req.body.name,
+                email: req.body.email,
+                passwordHash: req.body.passwordHash,
```

Comment:
req.body will contain password only
also de-structure or assign req.body in another variable and use it instead of repeating req.body